### PR TITLE
Feature/test package object model

### PR DESCRIPTION
**Pull request by mtrupkin**

### CODE REVIEW

File: client/src/main/java/tds/testpackage/model/Dependency.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ *  The Dependency can be read as:
+ *  if <ifToolType> is <ifToolCode>, then the option is enabled (or disabled)
+ *  e.g., "IF <Language> IS <ENU>, THEN <'Emboss' option should be disabled>"
+ */
+@AutoValue
+@JsonDeserialize(builder = AutoValue_Dependency.Builder.class)
+public abstract class Dependency {
+    public abstract String getIfToolType();
+    public abstract String getIfToolCode();
+    protected abstract Optional<String> getEnabled();
```
Comment (burzum619) [inline client/src/main/java/tds/testpackage/model/Dependency.java]: Should this be `Optional<Boolean>`?

File: client/src/main/java/tds/testpackage/model/XmlUtil.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+     *
+     * @param maybeValue xml value for a boolean
+     * @param defaultValue boolean value if xml value is not present
+     *
+     * @return translated java boolean value from xml
+     */
+    public static boolean parseBoolean(final Optional<String> maybeValue, boolean defaultValue) {
+        if (maybeValue.isPresent()) {
+            final String value = maybeValue.get();
+            switch (value) {
```
Comment (burzum619) [inline client/src/main/java/tds/testpackage/model/XmlUtil.java]: Can we add "true" or "false" as valid string values to parse?

File: service/src/main/java/tds/support/tool/repositories/MongoTestPackageRepository.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package tds.support.tool.repositories;
+
+import org.springframework.data.mongodb.repository.MongoRepository;
+
+import tds.testpackage.model.TestPackage;
+
```
Comment (burzum619) [inline service/src/main/java/tds/support/tool/repositories/MongoTestPackageRepository.java]: Can we add Javadocs here?

File: service/src/main/java/tds/support/tool/services/JobService.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package tds.support.tool.services;
 
+import java.io.IOException;
```
Comment (burzum619) [inline service/src/main/java/tds/support/tool/services/JobService.java]: Looks like we import this but not sure if we're using it in the class

File: client/src/main/java/tds/testpackage/model/XmlUtil.java (hunk 2 above)
Comment (mtrupkin) [reply client/src/main/java/tds/testpackage/model/XmlUtil.java]: Boolean.valueOf(value) will parse "true" and "false"

File: service/src/main/java/tds/support/tool/services/JobService.java (hunk 4 above)
Comment (mtrupkin) [reply service/src/main/java/tds/support/tool/services/JobService.java]: Agreed

**Pull request merged**